runtime/src/main/java/org/corfudb/protocols/wireprotocol/NettyCorfuMessageEncoder.java

```diff
@@ -1,29 +1,74 @@
 package org.corfudb.protocols.wireprotocol;
 
 import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufOutputStream;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.handler.codec.MessageToByteEncoder;
-
+import java.io.IOException;
+import java.util.concurrent.atomic.LongAccumulator;
 import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.proto.service.CorfuMessage.RequestMsg;
+import org.corfudb.runtime.proto.service.CorfuMessage.ResponseMsg;
 
-import java.util.concurrent.atomic.LongAccumulator;
+import static org.corfudb.protocols.CorfuProtocolCommon.LEGACY_CORFU_MSG_MARK;
+import static org.corfudb.protocols.CorfuProtocolCommon.PROTO_CORFU_REQUEST_MSG_MARK;
+import static org.corfudb.protocols.CorfuProtocolCommon.PROTO_CORFU_RESPONSE_MSG_MARK;
 
 /**
  * Created by mwei on 10/1/15.
  */
 @Slf4j
-public class NettyCorfuMessageEncoder extends MessageToByteEncoder<CorfuMsg> {
+public class NettyCorfuMessageEncoder extends MessageToByteEncoder<Object> {
```

Comment:
why object? Can it be a proto message?

Reply:
This is because it has to work for multiple Protobuf types: `RequestMsg` and `ResponseMsg`. And for the time being, it also has to work with legacy `CorfuMsg` types.